### PR TITLE
Improve robustness of command parsing

### PR DESCRIPTION
**Pull request by mweinelt**
This migrates the command parsing from an index and text splitting based
approach to a regular expression, making it more robust in the process.

Also renames parameter to argument, because parameters have a key/value
relationship while arguments do not.

---

The regular expression can be separated into three logical parts:
- the character class of command symbols, escaped using `re.escape`, so there will be no special meaning to characters like dot
- the match group command, that matches everything until the first whitespace
- the match group argument that matches everything after the first whitespace, defaults to `None` if not matched

We could also precompile the pattern, if anyone worries about performance.

<s>This change uses the walrus operator, introduced in Python 3.8. I couldn't find a supported version range, so I hope this is fine.</s>

Tested on my local instance and works as expected.

### CODE REVIEW

**Comment by azlux** [comment]
The current minimum python3 version is 3.6 because of f-string.
Did you face any issue with the simple split command for the need of regex implementation ?
(Changing parameter to arguments have a better mean, thanks)

**Comment by mweinelt** [comment]
This came up due to #271, but #272 was merged really quickly, so my revised commit (this one) arrived a bit late.

**Review by azlux — APPROVED** [review]
As discussed : 
- Stable debian python version is 3.7 : so the walrus operator have been remove
- The regex is clearer
- Have the same result than before but with a better code

File: mumbleBot.py [hunk 1]
```diff
@@ -252,20 +252,16 @@ def message_received(self, text):
...
-                command = message[0].lower()
-                parameter = ''
-                if len(message) > 1:
-                    parameter = message[1].rstrip()
-            else:
+        command_symbols = var.config.get('commands', 'command_symbol')
+        match = re.match(fr'^[{re.escape(command_symbols)}](?P<command>\S+)(?:\s(?P<argument>.*))?', message)
+        if match:
+            command = match.group("command").lower()
+            argument = match.group("argument")
```
Comment (mweinelt) [inline mumbleBot.py:259]:
```suggestion
            argument = match.group("argument") or ""
```

Which would make it default to empty string as expected by other parts of the codebase. See #278.

**Pull request merged**